### PR TITLE
New cobra-style commands for applications

### PR DESCRIPTION
**Pull request by jsuchome**
do not merge yet - not tested with real data..

### CODE REVIEW

**Review by stefannica — APPROVED** [review]
Looks great !

**Review by flaviodsr — CHANGES_REQUESTED** [review]
I think you do not need to expose the functions/types/attributes in the `application/*command*.go` files

File: pkg/cli/application/list.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		Run: func(cmd *cobra.Command, args []string) {
+			common.CheckErr(o.InitializeClients(c))
+			common.CheckErr(o.validate())
+			common.CheckErr(o.run())
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	cmd.Flags().StringVarP(&o.Type, "type", "t", "", "list only Applications of given type")
+	cmd.Flags().StringVarP(&o.Workflow, "workflow", "l", "", "list only Applications generated by given workflow")
```
Comment (flaviodsr) [inline pkg/cli/application/list.go:52]: insn't `w` instead of `l`?

File: pkg/cli/application/get.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+// NewSubCmdApplicationGet creates and returns the cobra command for the `application get` CLI command
+func NewSubCmdApplicationGet(gOpt *common.GlobalOptions) *cobra.Command {
+
+	o := NewGetOptions(gOpt)
+
+	cmd := &cobra.Command{
+		Use:   `get {-n|--name NAME}`,
+		Short: "Get an application.",
```
Comment (flaviodsr) [inline pkg/cli/application/get.go:34]: When referring to an application sometimes I see you using `application` and other times `Application`, better choose one and stick with it.

**Review by flaviodsr — CHANGES_REQUESTED** [review]

File: pkg/cli/application/list.go (hunk 1 above)
Comment (jsuchome) [reply pkg/cli/application/list.go:52]: yep

File: pkg/cli/application/get.go (hunk 2 above)
Comment (jsuchome) [reply pkg/cli/application/get.go:34]: ok

**Review by stefannica — APPROVED** [review]

**Pull request merged**